From anemo@mba.ocn.ne.jp  Tue Apr 21 15:55:07 2009
From: Atsushi Nemoto <anemo@mba.ocn.ne.jp>
Date: Thu, 16 Apr 2009 22:50:28 +0900
Subject: Driver core: platform: Do not complain for platform_data added by platform_device_add_data()
To: Ming Lei <tom.leiming@gmail.com>
Cc: Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <1239889828-7300-1-git-send-email-anemo@mba.ocn.ne.jp>


The commit ce21c7bc ("driver core: fix passing platform_data") added
an error message "foo: use which platform_data?", but this is normal
if platform_device_add_data() was used.

Do not print the error message if pdev->platform_data and
pdev->dev.platform_data have same value.

Signed-off-by: Atsushi Nemoto <anemo@mba.ocn.ne.jp>
Acked-by: Ming Lei <tom.leiming@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/platform.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/base/platform.c
+++ b/drivers/base/platform.c
@@ -253,7 +253,8 @@ int platform_device_add(struct platform_
 	* long time, so we allow the two cases coexist to make
 	* this kind of fix more easily*/
 	if (pdev->platform_data && pdev->dev.platform_data) {
-		printk(KERN_ERR
+		if (pdev->platform_data != pdev->dev.platform_data)
+			printk(KERN_ERR
 			       "%s: use which platform_data?\n",
 			       dev_name(&pdev->dev));
 	} else if (pdev->platform_data) {
